decentraland/unity-explorer
# fix: close livekit connection on logout

## What does this PR change?

Fixes #2225 

Several issues were found:
- The logout operation was missing the disconnection and re-connection of livekit.
- The livekit was not performing the handshake consistently after the logout. The livekit websocket was sharing the byte buffer pool with other classes. Since the handshake operation doesnt seem to be thread-safe it produced errors on sending the auth-chain when requesting the authentication, making the operation fail completely, thus not able to connect to the room & island.
- After the logout, the multiplayer systems were not able to run because the logout flow was not being completed. This was because a cancellation of the task was being requested wrongly. This prevented the profile propagation, thus the correct visualization of the remote avatar.
- The scene metadata in the `GateKeeperSceneRoom` was not being reset on disconnection. This prevented to re-connect to the scene after the logout.

## How to test the changes?

1. Follow issue steps
2. Avatar dissapear when the the user logs out
3. You should be able to keep seeing other avatars and send chat messages as expected
4. Perform `/goto` commands either to other scenes or worlds
5. You should be able to keep seeing other avatars and send chat messages as expected

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
